### PR TITLE
Fix news P3A enabled metric

### PR DESCRIPTION
**Pull request by DJAndries**
Resolves https://github.com/brave/brave-browser/issues/30709

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Follow plan in https://github.com/brave/brave-core/pull/18375

### CODE REVIEW

File: components/brave_news/browser/brave_news_controller.cc [hunk 1]
```diff
@@ -57,6 +58,9 @@ namespace {
 // The favicon size we desire. The favicons are rendered at 24x24 pixels but
 // they look quite a bit nicer if we get a 48x48 pixel icon and downscale it.
 constexpr uint32_t kDesiredFaviconSizePixels = 48;
+// Since we have two boolean prefs for the News enabled status, a delay
+// will be used so that we only report the histogram once for both pref updates.
```
Comment (petemill) [inline components/brave_news/browser/brave_news_controller.cc:62]: What's the problem with reporting the histogram twice - won't it just report only the most recent value for this metric?

Comment (DJAndries) [reply components/brave_news/browser/brave_news_controller.cc:62]: not quite, it seems that when a user presses "no thanks" on Android, both prefs will be enabled in the first call, and then the "show today" pref will be disabled in the second call. if we don't have the timer, the first call will set the "was enabled" pref to true which will result in the "is enabled" metric being reported as 0 in the second call, which we don't want. the timer is here to make sure all edge cases are covered and increases robustness

Comment (petemill) [reply components/brave_news/browser/brave_news_controller.cc:62]: This seems like a bug from Android. We should locate that part @deeppandya @tapanmodh ?

Comment (tapanmodh) [reply components/brave_news/browser/brave_news_controller.cc:61]: @petemill sure. I'll check it

**Comment by GeetaSarvadnya** [comment]
Verification INPROGRESS on

```
Brave | 1.55.5 Chromium: 115.0.5790.32 (Official Build) nightly (64-bit)
-- | --
Revision | f9b78cf410ecd5b372310ef25a12ef456c77d90f-refs/branch-heads/5790@{#724}
OS | Windows 10 Version 22H2 (Build 19045.3086)

```

**Pull request merged**